{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\USUARIO\\AppData\\Local\\Temp\\ipykernel_14968\\1875371614.py:1: DeprecationWarning: \n",
      "Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),\n",
      "(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)\n",
      "but was not found to be installed on your system.\n",
      "If this would cause problems for you,\n",
      "please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466\n",
      "        \n",
      "  import pandas as pd\n",
      "\n",
      "[notice] A new release of pip is available: 24.0 -> 24.2\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting GEOparseNote: you may need to restart the kernel to use updated packages.\n",
      "\n",
      "  Downloading GEOparse-2.0.4-py3-none-any.whl.metadata (6.5 kB)\n",
      "Requirement already satisfied: numpy>=1.7 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from GEOparse) (1.26.3)\n",
      "Requirement already satisfied: pandas>=0.17 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from GEOparse) (2.2.0)\n",
      "Requirement already satisfied: requests>=2.21.0 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from GEOparse) (2.32.3)\n",
      "Requirement already satisfied: tqdm>=4.31.1 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from GEOparse) (4.66.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from pandas>=0.17->GEOparse) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from pandas>=0.17->GEOparse) (2023.4)\n",
      "Requirement already satisfied: tzdata>=2022.7 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from pandas>=0.17->GEOparse) (2023.4)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from requests>=2.21.0->GEOparse) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from requests>=2.21.0->GEOparse) (3.7)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from requests>=2.21.0->GEOparse) (2.2.2)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from requests>=2.21.0->GEOparse) (2024.6.2)\n",
      "Requirement already satisfied: colorama in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from tqdm>=4.31.1->GEOparse) (0.4.6)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\usuario\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from python-dateutil>=2.8.2->pandas>=0.17->GEOparse) (1.16.0)\n",
      "Downloading GEOparse-2.0.4-py3-none-any.whl (29 kB)\n",
      "Installing collected packages: GEOparse\n",
      "Successfully installed GEOparse-2.0.4\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "%pip install GEOparse\n",
    "import GEOparse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "28-Aug-2024 13:55:55 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE136nnn/GSE136247/soft/GSE136247_family.soft.gz to ./tumoral-no-tumoral/GSE136247_family.soft.gz\n",
      "100%|██████████| 64.0M/64.0M [00:03<00:00, 18.9MB/s]  \n",
      "28-Aug-2024 13:55:59 DEBUG downloader - Size validation passed\n",
      "28-Aug-2024 13:55:59 DEBUG downloader - Moving C:\\Users\\USUARIO\\AppData\\Local\\Temp\\tmpy2yvu91d to d:\\d o c u\\PUCP\\TESIS\\Repo-HCC\\data-peru\\tumoral-no-tumoral\\GSE136247_family.soft.gz\n",
      "28-Aug-2024 13:55:59 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE136nnn/GSE136247/soft/GSE136247_family.soft.gz\n",
      "28-Aug-2024 13:55:59 INFO GEOparse - Parsing ./tumoral-no-tumoral/GSE136247_family.soft.gz: \n",
      "28-Aug-2024 13:55:59 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "28-Aug-2024 13:55:59 DEBUG GEOparse - SERIES: GSE136247\n",
      "28-Aug-2024 13:55:59 DEBUG GEOparse - PLATFORM: GPL17586\n",
      "c:\\Users\\USUARIO\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\GEOparse\\GEOparse.py:401: DtypeWarning: Columns (4,5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "28-Aug-2024 13:56:04 DEBUG GEOparse - SAMPLE: GSM4043276\n",
      "28-Aug-2024 13:56:04 DEBUG GEOparse - SAMPLE: GSM4043277\n",
      "28-Aug-2024 13:56:04 DEBUG GEOparse - SAMPLE: GSM4043278\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043279\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043280\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043281\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043282\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043283\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043284\n",
      "28-Aug-2024 13:56:05 DEBUG GEOparse - SAMPLE: GSM4043285\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043286\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043287\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043288\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043289\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043290\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043291\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043292\n",
      "28-Aug-2024 13:56:06 DEBUG GEOparse - SAMPLE: GSM4043293\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043294\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043295\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043296\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043297\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043298\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043299\n",
      "28-Aug-2024 13:56:07 DEBUG GEOparse - SAMPLE: GSM4043300\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043301\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043302\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043303\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043304\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043305\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043306\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043307\n",
      "28-Aug-2024 13:56:08 DEBUG GEOparse - SAMPLE: GSM4043308\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043309\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043310\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043311\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043312\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043313\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043314\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043315\n",
      "28-Aug-2024 13:56:09 DEBUG GEOparse - SAMPLE: GSM4043316\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043317\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043318\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043319\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043320\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043321\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043322\n",
      "28-Aug-2024 13:56:10 DEBUG GEOparse - SAMPLE: GSM4043323\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043324\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043325\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043326\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043327\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043328\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043329\n",
      "28-Aug-2024 13:56:11 DEBUG GEOparse - SAMPLE: GSM4043330\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043331\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043332\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043333\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043334\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043335\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043336\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043337\n",
      "28-Aug-2024 13:56:12 DEBUG GEOparse - SAMPLE: GSM4043338\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043339\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043340\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043341\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043342\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043343\n",
      "28-Aug-2024 13:56:13 DEBUG GEOparse - SAMPLE: GSM4043344\n"
     ]
    }
   ],
   "source": [
    "df_tumoral_notumoral = GEOparse.get_GEO(\"GSE136247\",destdir=\"./tumoral-no-tumoral/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'title': ['Expression data of tumor and non-tumor liver tissues from Peruvian patients with hepatocellular carcinoma (HCC)'],\n",
       " 'geo_accession': ['GSE136247'],\n",
       " 'status': ['Public on Jan 31 2021'],\n",
       " 'submission_date': ['Aug 23 2019'],\n",
       " 'last_update_date': ['Mar 23 2021'],\n",
       " 'pubmed_id': ['33747361'],\n",
       " 'web_link': ['https://www.oncotarget.com/article/27890/text/'],\n",
       " 'summary': ['A large fraction of HCC in Peru arises in younger, non-cirrhotic patients. Transcriptomic analysis of the HCC and non-tumor liver (NTL) tissues highlights a divergent gene expression profile that coincides with Amerind haplogrouping. This metagene is also associated with a peculiar molecular signature that does not conform with the current molecular classifications of HCCs.',\n",
       "  'We used microarrays to compare gene expression between tumors and non-tumor counter parts in order to identify genes and ontologies that are differentially expressed.'],\n",
       " 'overall_design': ['HCC patients were treated by anatomic liver resection. Non-tumor liver tissues were obtained from tumor-free margins of resected surgical pieces. Total RNA was extracted from flash-frozen samples for hybridization on Affymetrix microarrays.'],\n",
       " 'type': ['Expression profiling by array'],\n",
       " 'contributor': ['Juan,P,Cerapio', 'Pascal,,Pineau'],\n",
       " 'sample_id': ['GSM4043276',\n",
       "  'GSM4043277',\n",
       "  'GSM4043278',\n",
       "  'GSM4043279',\n",
       "  'GSM4043280',\n",
       "  'GSM4043281',\n",
       "  'GSM4043282',\n",
       "  'GSM4043283',\n",
       "  'GSM4043284',\n",
       "  'GSM4043285',\n",
       "  'GSM4043286',\n",
       "  'GSM4043287',\n",
       "  'GSM4043288',\n",
       "  'GSM4043289',\n",
       "  'GSM4043290',\n",
       "  'GSM4043291',\n",
       "  'GSM4043292',\n",
       "  'GSM4043293',\n",
       "  'GSM4043294',\n",
       "  'GSM4043295',\n",
       "  'GSM4043296',\n",
       "  'GSM4043297',\n",
       "  'GSM4043298',\n",
       "  'GSM4043299',\n",
       "  'GSM4043300',\n",
       "  'GSM4043301',\n",
       "  'GSM4043302',\n",
       "  'GSM4043303',\n",
       "  'GSM4043304',\n",
       "  'GSM4043305',\n",
       "  'GSM4043306',\n",
       "  'GSM4043307',\n",
       "  'GSM4043308',\n",
       "  'GSM4043309',\n",
       "  'GSM4043310',\n",
       "  'GSM4043311',\n",
       "  'GSM4043312',\n",
       "  'GSM4043313',\n",
       "  'GSM4043314',\n",
       "  'GSM4043315',\n",
       "  'GSM4043316',\n",
       "  'GSM4043317',\n",
       "  'GSM4043318',\n",
       "  'GSM4043319',\n",
       "  'GSM4043320',\n",
       "  'GSM4043321',\n",
       "  'GSM4043322',\n",
       "  'GSM4043323',\n",
       "  'GSM4043324',\n",
       "  'GSM4043325',\n",
       "  'GSM4043326',\n",
       "  'GSM4043327',\n",
       "  'GSM4043328',\n",
       "  'GSM4043329',\n",
       "  'GSM4043330',\n",
       "  'GSM4043331',\n",
       "  'GSM4043332',\n",
       "  'GSM4043333',\n",
       "  'GSM4043334',\n",
       "  'GSM4043335',\n",
       "  'GSM4043336',\n",
       "  'GSM4043337',\n",
       "  'GSM4043338',\n",
       "  'GSM4043339',\n",
       "  'GSM4043340',\n",
       "  'GSM4043341',\n",
       "  'GSM4043342',\n",
       "  'GSM4043343',\n",
       "  'GSM4043344'],\n",
       " 'contact_name': ['Juan Pablo,,Cerapio'],\n",
       " 'contact_email': ['jcerapioarroyo@gmail.com, juan-pablo.cerapio-arroyo@inserm.fr'],\n",
       " 'contact_institute': ['CRCT'],\n",
       " 'contact_address': ['Av 2 Hubert Curien'],\n",
       " 'contact_city': ['Toulouse'],\n",
       " 'contact_zip/postal_code': ['31000'],\n",
       " 'contact_country': ['France'],\n",
       " 'supplementary_file': ['ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE136nnn/GSE136247/suppl/GSE136247_RAW.tar'],\n",
       " 'platform_id': ['GPL17586'],\n",
       " 'platform_taxid': ['9606'],\n",
       " 'sample_taxid': ['9606'],\n",
       " 'relation': ['BioProject: https://www.ncbi.nlm.nih.gov/bioproject/PRJNA561755']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tumoral_notumoral.metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tGSM4043276 transferred\n",
      "\n",
      "\t\tGSM4043277 transferred\n",
      "\n",
      "\t\tGSM4043278 transferred\n",
      "\n",
      "\t\tGSM4043279 transferred\n",
      "\n",
      "\t\tGSM4043280 transferred\n",
      "\n",
      "\t\tGSM4043281 transferred\n",
      "\n",
      "\t\tGSM4043282 transferred\n",
      "\n",
      "\t\tGSM4043283 transferred\n",
      "\n",
      "\t\tGSM4043284 transferred\n",
      "\n",
      "\t\tGSM4043285 transferred\n",
      "\n",
      "\t\tGSM4043286 transferred\n",
      "\n",
      "\t\tGSM4043287 transferred\n",
      "\n",
      "\t\tGSM4043288 transferred\n",
      "\n",
      "\t\tGSM4043289 transferred\n",
      "\n",
      "\t\tGSM4043290 transferred\n",
      "\n",
      "\t\tGSM4043291 transferred\n",
      "\n",
      "\t\tGSM4043292 transferred\n",
      "\n",
      "\t\tGSM4043293 transferred\n",
      "\n",
      "\t\tGSM4043294 transferred\n",
      "\n",
      "\t\tGSM4043295 transferred\n",
      "\n",
      "\t\tGSM4043296 transferred\n",
      "\n",
      "\t\tGSM4043297 transferred\n",
      "\n",
      "\t\tGSM4043298 transferred\n",
      "\n",
      "\t\tGSM4043299 transferred\n",
      "\n",
      "\t\tGSM4043300 transferred\n",
      "\n",
      "\t\tGSM4043301 transferred\n",
      "\n",
      "\t\tGSM4043302 transferred\n",
      "\n",
      "\t\tGSM4043303 transferred\n",
      "\n",
      "\t\tGSM4043304 transferred\n",
      "\n",
      "\t\tGSM4043305 transferred\n",
      "\n",
      "\t\tGSM4043306 transferred\n",
      "\n",
      "\t\tGSM4043307 transferred\n",
      "\n",
      "\t\tGSM4043308 transferred\n",
      "\n",
      "\t\tGSM4043309 transferred\n",
      "\n",
      "\t\tGSM4043310 transferred\n",
      "\n",
      "\t\tGSM4043311 transferred\n",
      "\n",
      "\t\tGSM4043312 transferred\n",
      "\n",
      "\t\tGSM4043313 transferred\n",
      "\n",
      "\t\tGSM4043314 transferred\n",
      "\n",
      "\t\tGSM4043315 transferred\n",
      "\n",
      "\t\tGSM4043316 transferred\n",
      "\n",
      "\t\tGSM4043317 transferred\n",
      "\n",
      "\t\tGSM4043318 transferred\n",
      "\n",
      "\t\tGSM4043319 transferred\n",
      "\n",
      "\t\tGSM4043320 transferred\n",
      "\n",
      "\t\tGSM4043321 transferred\n",
      "\n",
      "\t\tGSM4043322 transferred\n",
      "\n",
      "\t\tGSM4043323 transferred\n",
      "\n",
      "\t\tGSM4043324 transferred\n",
      "\n",
      "\t\tGSM4043325 transferred\n",
      "\n",
      "\t\tGSM4043326 transferred\n",
      "\n",
      "\t\tGSM4043327 transferred\n",
      "\n",
      "\t\tGSM4043328 transferred\n",
      "\n",
      "\t\tGSM4043329 transferred\n",
      "\n",
      "\t\tGSM4043330 transferred\n",
      "\n",
      "\t\tGSM4043331 transferred\n",
      "\n",
      "\t\tGSM4043332 transferred\n",
      "\n",
      "\t\tGSM4043333 transferred\n",
      "\n",
      "\t\tGSM4043334 transferred\n",
      "\n",
      "\t\tGSM4043335 transferred\n",
      "\n",
      "\t\tGSM4043336 transferred\n",
      "\n",
      "\t\tGSM4043337 transferred\n",
      "\n",
      "\t\tGSM4043338 transferred\n",
      "\n",
      "\t\tGSM4043339 transferred\n",
      "\n",
      "\t\tGSM4043340 transferred\n",
      "\n",
      "\t\tGSM4043341 transferred\n",
      "\n",
      "\t\tGSM4043342 transferred\n",
      "\n",
      "\t\tGSM4043343 transferred\n",
      "\n",
      "\t\tGSM4043344 transferred\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for name, extra in df_tumoral_notumoral.gsms.items():\n",
    "    name = name.strip('\\n')\n",
    "    print('\\t\\t'+name+' transferred\\n')\n",
    "    output_path = os.path.join(\"./tumoral-no-tumoral/\", name + '.txt')\n",
    "    df_tumoral_notumoral.gsms[name].table.to_csv(output_path, index = None, sep = '\\t', mode = 'w')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_notumoral = GEOparse.get_GEO(\"GSE111580\",destdir=\"./no-tumoral/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_notumoral.metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for name, extra in df_tumoral_notumoral.gsms.items():\n",
    "    name = name.strip('\\n')\n",
    "    print('\\t\\t'+name+' transferred\\n')\n",
    "    output_path = os.path.join(\"./no-tumoral/\", name + '.txt')\n",
    "    df_tumoral_notumoral.gsms[name].table.to_csv(output_path, index = None, sep = '\\t', mode = 'w')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
